### PR TITLE
Thuang feedback seamless

### PR DESCRIPTION
**Pull request by tihuan**
#### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks Timmy!

File: client/__tests__/e2e/puppeteer.setup.ts [hunk 1]
```diff
@@ -29,6 +29,11 @@ beforeEach(async () => {
         status: 200,
         contentType: "application/json",
         body: JSON.stringify(DATASET_METADATA_RESPONSE),
+        // (thuang): Add headers so FE@localhost:3000 can access API@localhost:5000
+        headers: {
+          "Access-Control-Allow-Origin": "http://localhost:3000",
```
Comment (tihuan) [inline client/__tests__/e2e/puppeteer.setup.ts:34]: Added these headers 🙆‍♂️

**Comment by tihuan** [comment]
Made some tiny changes 💡 PTAL again, @MillenniumFalconMechanic 🙏 Thanks so much!

File: client/__tests__/e2e/puppeteer.setup.ts (hunk 1 above)
Comment (MillenniumFalconMechanic) [reply client/__tests__/e2e/puppeteer.setup.ts:34]: Thanks Timmy!  :ok_woman:

File: client/src/actions/index.ts [hunk 2]
```diff
@@ -81,9 +81,10 @@ async function datasetMetadataFetch(
   const datasetMetadata = await fetchJson<{ metadata: DatasetMetadata }>(
     "dataset-metadata"
   );
+
   dispatch({
     type: "dataset metadata load complete",
-    datasetMetadata,
+    datasetMetadata: datasetMetadata.metadata,
```
[the remark this answers is not in the document]
Comment (MillenniumFalconMechanic) [reply client/src/actions/index.ts:87]: Interesting. I'll take a look at why this passed for me locally.

[the remark this answers is not in the document]
Comment (tihuan) [reply client/src/actions/index.ts:87]: Did we find anything regarding `.metadata` namespace 😄 ? Thank you!

File: client/__tests__/e2e/puppeteer.setup.ts (hunk 1 above)
Comment (MillenniumFalconMechanic) [reply client/__tests__/e2e/puppeteer.setup.ts:34]: How do you run your tests locally? I am not seeing these errors when running `make smoke-test`.

**Pull request merged**